src/data/nat/basic.lean

```diff
@@ -263,6 +263,9 @@ end
 @[simp] lemma pred_one_add (n : ℕ) : pred (1 + n) = n :=
 by rw [add_comm, add_one, pred_succ]
 
+@[simp] lemma succ_pred_eq_of_ne_zero (n : ℕ) (h : n ≠ 0) : n.pred.succ = n :=
```

Comment:
We already have `succ_pred_eq_of_pos`, which you can use in the proof (or might decide makes this lemma irrelevant). Probably should not be a simp lemma.

Reply:
This would not be a simp lemma because it would be difficult to synthesize the `ne` hypothesis, correct?